userspace/libsinsp/state/table_adapters.h

```diff
@@ -47,6 +47,92 @@ class fixed_dynamic_fields_infos : public dynamic_struct::field_infos {
 	}
 };
 
+/**
+ * @brief An adapter for the libsinsp::state::table_entry interface
+ * that wraps a non-owning pointer of arbitrary pair of type T. The underlying pointer
+ * can be set and unset arbitrarily, making this wrapper suitable for optimized
+ * allocations. Instances of table_entry from this adapter have no static fields,
+ * and make the wrapped value available as a single dynamic field. The dynamic
+ * fields definitions of this wrapper are fixed and immutable.
+ */
+template<typename Tfirst, typename Tsecond>
+class pair_table_entry_adapter : public libsinsp::state::table_entry {
+public:
+	// note: this dynamic definitions are fixed in size and structure,
+	// so there's no need of worrying about specific identifier checks
+	// as they should be safely interchangeable
+	static const constexpr uintptr_t s_dynamic_fields_id = 4321;
+
+	struct dynamic_fields_t : public fixed_dynamic_fields_infos {
+		using _dfi = dynamic_struct::field_info;
+
+		inline dynamic_fields_t():
+		        fixed_dynamic_fields_infos(
```

Comment:
Fields will be called `pair.first` and `pair.second`.

Reply:
IMHO you should rename these fields to `first` and `second` to make them simpler :)
I see no other possible fields the pair type, and even if we want to add other fields, I see no reason why we should categorize them by putting a prefix before the name.

Reply:
I don't like the fact that we have a generic `value` and then we would have `first` and `second`.
Another solution would be `value.first` and `value.second`; indeed, i like the categorization, from an UX perspective it seems easier to guess for the developer.
I have no strong opinions though, can easily change it if you all agree on just `first` and `second` :D @jasondellaluce